perf/perf_metric.py

```diff
@@ -111,6 +116,39 @@ def test_all_metric_events_with_M(self):
     def test_all_metric_events_with_metric(self):
         self._run_cmd("--metric-only -M")
 
+    def test_cpi_stall(self):
+        """
+        Function to test CPI_STALL_RATIO feature. Power10 PMU provides events
+        to understand stall cycles of different pipeline stages.
+        """
+        # Check if CPI_STALL_RATIO metricgroup is present in perf list or not
+        output = process.system_output('perf list metricgroup')
+        if self.rev == '0080' and 'CPI_STALL_RATIO' in output.decode().split():
+            cmd = "perf stat --metric-no-group -M CPI_STALL_RATIO perf bench sched messaging"
+            res = process.run(cmd, shell=True, verbose=True)
+            result = (res.stdout + res.stderr).decode()
+            if res.exit_status != 0 or "not counted" in result or "not supported" in result:
+                self.fail("Failed to collect perf stats with CPI_STALL_RATIO"
+                          " metricgroup")
+
+            # Check if the required metrics are present and only once
+            required_metrics = ['PM_CMPL_STALL', 'PM_ISSUE_STALL', 'PM_RUN_INST_CMPL',
+                                'PM_DISP_STALL_CYC', 'PM_EXEC_STALL',
+                                'COMPLETION_STALL_CPI', 'ISSUE_STALL_CPI',
+                                'DISPATCH_STALL_CPI', 'EXECUTION_STALL_CPI']
```

Comment:
Can these matrices also drive from YAML as if any new Feature added

Reply:
I feel yaml option is not needed here. We will check for these metrics only for this feature.
Still you want I can change code to yaml based.